pkg/provider/onboardbase/client/client.go

```diff
@@ -0,0 +1,346 @@
+	OnboardbaseAPIKey string
+	VerifyTLS    bool
+	UserAgent    string
+	OnboardbasePassCode string
+}
+
+type queryParams map[string]string
+
+type headers map[string]string
+
+type httpRequestBody []byte
+
+type Secrets map[string]string
+
+type RawSecret struct {
+	Key string `json:"key,omitempty"`
+	Value string `json:"value,omitempty"`
+}
+
+type RawSecrets []RawSecret
+
+type APIError struct {
+	Err     error
+	Message string
+	Data    string
+}
+
+type apiResponse struct {
+	HTTPResponse *http.Response
+	Body         []byte
+}
+
+type apiErrorResponse struct {
+	Messages []string
+	Success  bool
+}
+
+type SecretRequest struct {
+	Environment    string
+	Project string
+	Name    string
+}
+
+type SecretsRequest struct {
+	Environment    string
+	Project string
+}
+
+type UpdateSecretsRequest struct {
+	Secrets RawSecrets `json:"secrets,omitempty"`
+	Project string     `json:"project,omitempty"`
+	Config  string     `json:"config,omitempty"`
+}
+
+type secretResponseBodyObject struct {
+	Title string `json:"title,omitempty"`
+	Id string `json:"id,omitempty"`
+}
+
+type secretResponseBodyData struct {
+	Project secretResponseBodyObject `json:"project,omitempty"`
+	Environment secretResponseBodyObject `json:"environment,omitempty"`
+	Team secretResponseBodyObject `json:"team,omitempty"`
+	Secrets []string `json:"secrets,omitempty"`
+}
+
+type secretResponseBody struct {
+	Data secretResponseBodyData `json:"data,omitempty"`
+	Message string `json:"message,omitempty"`
+	Status string `json:"status,omitempty"`
+}
+
+type SecretResponse struct {
+	Name  string
+	Value string
+}
+
+type SecretsResponse struct {
+	Secrets  Secrets
+	Body     []byte
+}
+
+func NewOnboardbaseClient(onboardbaseAPIKey, onboardbasePasscode string) (*OnboardbaseClient, error) {
+	client := &OnboardbaseClient{
+		OnboardbaseAPIKey: onboardbaseAPIKey,
+		OnboardbasePassCode: onboardbasePasscode,
+		VerifyTLS:    true,
+		UserAgent:    "onboardbase-external-secrets",
+	}
+
+	if err := client.SetBaseURL("https://public.onboardbase.com/api/v1/"); err != nil {
+		return nil, &APIError{Err: err, Message: "setting base URL failed"}
+	}
+
+	return client, nil
+}
+
+func (c *OnboardbaseClient) BaseURL() *url.URL {
+	u := *c.baseURL
+	return &u
+}
+
+func (c *OnboardbaseClient) SetBaseURL(urlStr string) error {
+	baseURL, err := url.Parse(strings.TrimSuffix(urlStr, "/"))
+
+	if err != nil {
+		return err
+	}
+
+	if baseURL.Scheme == "" {
+		baseURL.Scheme = "https"
+	}
+
+	c.baseURL = baseURL
+	return nil
+}
+
+func (c *OnboardbaseClient) Authenticate() error {
+
+	if _, err := c.performRequest("/team/members", "GET", headers{}, queryParams{	}, httpRequestBody{}); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (c *OnboardbaseClient) getSecretsFromPayload(data secretResponseBodyData) (map[string]string, error) {
+	kv := make(map[string]string)
+	for _, secret := range data.Secrets {
+		passphrase := c.OnboardbasePassCode
+		decrypted := aesdecrypt.Run(secret, passphrase)
```

Comment:
This possible does a `log.Fatal()` which breaks the controller, could you fix that upstream and return an error instead?

wrt: https://github.com/Onboardbase/go-cryptojs-aes-decrypt/blob/main/decrypt/aesdecrypt.go#L35-L39

Reply:
Thank you